docs/source/developers/guide/step_by_step/pr_and_github.rst

```diff
@@ -24,14 +24,168 @@
 
 .. _pr_and_github:
 
-****************************
-Lifecycle of a Pull Request
-****************************
+******************************
+Lifecycle of a Pull Request 🙀 
+******************************
 
+:ref:`As mentioned before<set-up>`, the Arrow project uses git for
+version control and a workflow based on Pull Requests. That means
+that you contribute the changes to the code by creating a branch
+in Git, make changes to the code, push the changes to your ``origin``
+which is your fork of the Arrow repository on GitHub and then you
+create a **Pull Request** against the official Arrow repository
+which is saved in your set up as ``upstream``.
 
-Creating a PR 🙀
-================
+You should have git set up by now, have cloned the repository,
+have successfully built Arrow and have an JIRA issue to work on.
 
+**Before making changes to the code, you should create a new
+branch in Git.**
 
-Reviews and get the PR merge 🎉
-===============================
\ No newline at end of file
+1. Update/sync the code from your ``upstream``
+   in the master branch. Run it in the shell from ``arrow`` directory.
+
+   .. code:: console
+
+      $ git checkout master
```

Comment:
@jorisvandenbossche here for example, `$` gets copy-pasted also if I try it in the `index.html` view of the guide.

Reply:
Which version of sphinx are you using locally to build the docs?

Reply:
(I also don't know if it matters whether you use `.. code::` vs `.. code-block::`, not directly sure what's the difference between both)

Reply:
sphinx-build 2.4.4

I tried `.. code::` and `.. code-block::` and was the same. I am quite sure it's just my setup.

Reply:
That's a very old version. I suppose that's the reason for this not working. I think you need sphinx >= 4

Reply:
We should update [requirements.txt](https://github.com/apache/arrow/blob/master/docs/requirements.txt). Why is there a set version anyways?

Reply:
I deleted the pin for Sphinx in `requirements.txt` and built the docs with 4.3.1 version successfully. Only warnings for some compute functions persist. I will add the change as a PR to the [ARROW-8340](https://issues.apache.org/jira/browse/ARROW-8340).